crates/iroha_genesis/src/lib.rs

```diff
@@ -357,33 +331,74 @@ impl Decode for ExecutorPath {
     }
 }
 
+impl From<PathBuf> for ExecutorPath {
+    fn from(value: PathBuf) -> Self {
+        Self(value)
+    }
+}
+
+impl TryFrom<ExecutorPath> for Executor {
+    type Error = eyre::Report;
+
+    fn try_from(value: ExecutorPath) -> Result<Self, Self::Error> {
+        let wasm = fs::read(&value.0)
+            .wrap_err_with(|| eyre!("failed to read the executor from {}", value.0.display()))?;
+
+        Ok(Executor::new(WasmSmartContract::from_compiled(wasm)))
+    }
+}
+
+impl ExecutorPath {
+    /// Resolve `self` to `here/self`,
+    /// assuming `self` is an unresolved relative path to `here`.
+    /// Must be applied once.
+    fn resolve(&mut self, here: impl AsRef<Path>) {
+        self.0 = here.as_ref().join(&self.0)
+    }
+}
```

Comment:
To minimize impact to schema, I didn't introduce rich objects like `iroha_swarm::path::{AbsolutePath, RelativePath}` and `iroha_config_base::WithOrigin`